desc/equilibrium/coords.py

```diff
@@ -651,67 +652,56 @@ def to_sfl(
     M_grid = M_grid or int(2 * M)
     N_grid = N_grid or int(2 * N)
 
-    grid = ConcentricGrid(L_grid, M_grid, N_grid, node_pattern="ocs", NFP=eq.NFP)
-    bdry_grid = LinearGrid(M=M, N=N, rho=1.0, NFP=eq.NFP)
+    eq_PEST = eq.copy() if copy else eq
+    eq_PEST.change_resolution(L, M, N, NFP=1)
 
-    toroidal_coords = eq.compute(["R", "Z", "lambda"], grid=grid)
-    theta = grid.nodes[:, 1]
-    vartheta = theta + toroidal_coords["lambda"]
-    sfl_grid = Grid(np.array([grid.nodes[:, 0], vartheta, grid.nodes[:, 2]]).T)
-
-    bdry_coords = eq.compute(["R", "Z", "lambda"], grid=bdry_grid)
-    bdry_theta = bdry_grid.nodes[:, 1]
-    bdry_vartheta = bdry_theta + bdry_coords["lambda"]
-    bdry_sfl_grid = Grid(
-        np.array([bdry_grid.nodes[:, 0], bdry_vartheta, bdry_grid.nodes[:, 2]]).T
+    grid_PEST = ConcentricGrid(L_grid, M_grid, N_grid, node_pattern="ocs", NFP=1)
+    data = eq.compute(
+        ["R", "Z", "lambda"],
+        Grid(map_coordinates(eq, grid_PEST.nodes, ("rho", "theta_PEST", "zeta"))),
     )
 
-    if copy:
-        eq_sfl = eq.copy()
-    else:
-        eq_sfl = eq
-    eq_sfl.change_resolution(L, M, N)
+    eq_PEST.R_lmn = Transform(
+        grid_PEST,
+        eq_PEST.R_basis,
+        build=False,
+        build_pinv=True,
+        rcond=rcond,
+    ).fit(data["R"])
 
-    R_sfl_transform = Transform(
-        sfl_grid, eq_sfl.R_basis, build=False, build_pinv=True, rcond=rcond
-    )
-    R_lmn_sfl = R_sfl_transform.fit(toroidal_coords["R"])
-    del R_sfl_transform  # these can take up a lot of memory so delete when done.
+    eq_PEST.Z_lmn = Transform(
+        grid_PEST,
+        eq_PEST.Z_basis,
+        build=False,
+        build_pinv=True,
+        rcond=rcond,
+    ).fit(data["Z"])
+
+    eq_PEST.L_lmn = np.zeros_like(eq_PEST.L_lmn)
```

Comment:
I retained this logic from `master` for line 680, but I do not understand. Is `eq_PEST.L_lmn` initialized to something other than zero?